apps/nerves_hub_web_core/lib/nerves_hub_web_core/firmwares/firmware_metadata.ex

```diff
@@ -38,6 +38,19 @@ defmodule NervesHubWebCore.Firmwares.FirmwareMetadata do
           version: Version.build()
         }
 
+  @type metadata :: %{
+          optional(:architecture) => String.t(),
+          optional(:author) => String.t() | nil,
+          optional(:description) => String.t() | nil,
+          optional(:fwup_version) => Version.build() | nil,
+          optional(:misc) => String.t() | nil,
+          optional(:platform) => String.t(),
+          optional(:product) => String.t(),
+          optional(:uuid) => Ecto.UUID.t(),
+          optional(:vcs_identifier) => String.t() | nil,
+          optional(:version) => Version.build()
+        }
```

Comment:
```suggestion
  @type metadata :: %{
          :architecture => String.t(),
          :author => String.t() | nil,
          :description => String.t() | nil,
          optional(:fwup_version) => Version.build() | nil,
          :misc => String.t() | nil,
          :platform => String.t(),
          :product => String.t(),
          :uuid => Ecto.UUID.t(),
          :vcs_identifier => String.t() | nil,
          :version => Version.build()
        }
```

Also, for simplicity, this should work as well. `fwup_version` only comes in the device channel and not from the firmware DB struct or fwup, therefor it is "optional"